lms/djangoapps/courseware/courses.py

```diff
@@ -104,12 +133,11 @@ def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=
         raise CoursewareAccessException(access_response)
 
     if check_if_enrolled:
-        # Verify that the user is either enrolled in the course or a staff member.
-        # If user is not enrolled, raise UserNotEnrolled exception that will be caught by middleware.
-        if not ((user.id and CourseEnrollment.is_enrolled(user, course_key)) or has_access(user, 'staff', course)):
-            raise UserNotEnrolled(course_key)
-
-    return course
```

Comment:
It's an illusion from the diff.  I introduced a new method `check_course_access` that does only the access verification, taking in either a `CourseOverview` or a `CourseDescriptor` and not returning any values.  The pre-existing `get_course_with_access` function and the new `get_course_overview_with_access` function return the course.